shopify/collection.py

```diff
@@ -0,0 +1,143 @@
+from pyactiveresource.collection import Collection
+from six.moves.urllib.parse import urlparse, parse_qs
+import cgi
+
+class PaginatedCollection(Collection):
+    """
+    A subclass of Collection which allows cycling through pages of
+    data through cursor-based pagination.
+
+    :next_page_url contains a url for fetching the next page
+    :previous_page_url contains a url for fetching the previous page
+
+    You can use next_page_url and previous_page_url to fetch the next page
+    of data by calling Resource.find(from_=page.next_page_url)
+    """
+
+    def __init__(self, *args, **kwargs):
+        """If given a Collection object as an argument, inherit its metadata."""
+
+        metadata = kwargs.pop("metadata", None)
+        obj = args[0]
+        if isinstance(obj, Collection):
+            if metadata:
+                metadata.update(obj.metadata)
+            else:
+                metadata = obj.metadata
+            super(PaginatedCollection, self).__init__(obj, metadata=metadata)
+        else:
+            super(PaginatedCollection, self).__init__(metadata=metadata or {},
+                                                      *args, **kwargs)
+        if not ("pagination" in self.metadata and "resource_class" in self.metadata):
+            raise AttributeError("Cursor-based pagination requires \"pagination\" and \"resource_class\" attributes in the metadata.")
+
+        self.next_page_url = self.metadata["pagination"].get('next', None)
+        self.previous_page_url = self.metadata["pagination"].get('previous', None)
+
+        self._next = None
+        self._previous = None
+        self._current_iter = None
+        self._no_iter_next = kwargs.pop("no_iter_next", False)
+
+    def has_previous(self):
+        """Returns true if the current page has any previous pages before it.
+        """
+        return bool(self.previous_page_url)
+
+    def has_next(self):
+        """Returns true if the current page has any pages beyond the current position.
+        """
+        return bool(self.next_page_url)
+
+    def previous(self, no_cache=False):
```

Comment:
Nitpick: Imo `next_page` and `previous_page` are perhaps more clear than just `next` and `previous`.

Reply:
Thanks, I had the same sort of feeling so it helps to here someone else is thinking the same. I will make that change.